Core/src/Core/Filters/CalculateArrayHistogramFilter.cpp

```diff
@@ -0,0 +1,155 @@
+  params.insert(std::make_unique<MultiArraySelectionParameter>(k_SelectedArrayPaths_Key, "DataArray(s) to Histogram", "", MultiArraySelectionParameter::ValueType{}, complex::GetAllNumericTypes()));
+  params.insert(std::make_unique<DataGroupCreationParameter>(k_NewDataGroupName_Key, "New DataGroup Path", "", DataPath{}));
+  params.insert(std::make_unique<DataGroupSelectionParameter>(k_DataGroupName_Key, "Output DataGroup Path", "", DataPath{}));
+  // Associate the Linkable Parameter(s) to the children parameters that they control
+  params.linkParameters(k_UserDefinedRange_Key, k_MinRange_Key, true);
+  params.linkParameters(k_UserDefinedRange_Key, k_MaxRange_Key, true);
+  params.linkParameters(k_NewDataGroup_Key, k_NewDataGroupName_Key, true);
+  params.linkParameters(k_NewDataGroup_Key, k_DataGroupName_Key, false);
+
+  return params;
+}
+
+//------------------------------------------------------------------------------
+IFilter::UniquePointer CalculateArrayHistogramFilter::clone() const
+{
+  return std::make_unique<CalculateArrayHistogramFilter>();
+}
+
+//------------------------------------------------------------------------------
+IFilter::PreflightResult CalculateArrayHistogramFilter::preflightImpl(const DataStructure& dataStructure, const Arguments& filterArgs, const MessageHandler& messageHandler,
+                                                                      const std::atomic_bool& shouldCancel) const
+{
+  auto pNumberOfBinsValue = filterArgs.value<int32>(k_NumberOfBins_Key);
+  auto pUserDefinedRangeValue = filterArgs.value<bool>(k_UserDefinedRange_Key); // verify and calculate range values here if false
+  auto pMinRangeValue = filterArgs.value<float64>(k_MinRange_Key);
+  auto pMaxRangeValue = filterArgs.value<float64>(k_MaxRange_Key);
+  auto pNewDataGroupValue = filterArgs.value<bool>(k_NewDataGroup_Key);
+  auto pDataGroupNameValue = filterArgs.value<DataPath>(k_DataGroupName_Key);
+  auto pSelectedArrayPathsValue = filterArgs.value<MultiArraySelectionParameter::ValueType>(k_SelectedArrayPaths_Key);
+  auto pNewDataGroupNameValue = filterArgs.value<DataPath>(k_NewDataGroupName_Key); // sanity check if is Attribute matrix after impending complex update
+
+  PreflightResult preflightResult;
+
+  complex::Result<OutputActions> resultOutputActions;
+
+  std::vector<PreflightValue> preflightUpdatedValues;
+
+  if(pNewDataGroupValue)
+  {
+    auto createDataGroupAction = std::make_unique<CreateDataGroupAction>(pNewDataGroupNameValue);
+    resultOutputActions.value().actions.push_back(std::move(createDataGroupAction));
+  }
+  for(auto& selectedArrayPath : pSelectedArrayPathsValue)
+  {
+    const auto& dataArray = dataStructure.getDataAs<IDataArray>(selectedArrayPath);
+    DataPath childPath;
+    if(pNewDataGroupValue)
+    {
+      childPath = pNewDataGroupNameValue.createChildPath((dataArray->getName() + "Histogram"));
```

Comment:
A few things: 
*IF* we do not allow the user to actually set that suffix then make it a string constant at the top of the file and probably put a space at the front of the string so that the output arrays are a bit more readable.

I would strongly advocate adding in another parameter to allow the user to set that suffix to what they want.